### DISCUSSION TITLE
Help make Obsidian Kanban better

### DISCUSSION THREAD

**Discussion by mgmeyers** [mgmeyers/obsidian-kanban#5] [Ideas & Feature Requests]
Greetings!

I could use your help making Obsidian Kanban better and more useful for everyone.

First, if you encounter any bugs, please report them here: https://github.com/mgmeyers/obsidian-kanban/issues

But more importantly, tell me how you use, or would like to use `obsidian-kanban`:

- What features are missing or incompatible with your desired workflow? 
- What are points of friction?
- What would you like the resulting markdown to look like?

**Comment by jgsource** [comment]
Since I use kanban as my to do list, I would like the ability to checklist tasks and even have the option to make them disappear(but still show in markdown) after getting checked off. I know I can just have a section labeled as done, but I currently have 3 lists in one kanban to deal with different types of tasks.

**Reply by mgmeyers** [reply]
@jgsource There is now a setting to show checkboxes. You can also cmd/ctrl + click on the checkbox to archive a card, which makes them only visible in markdown mode.

**Reply by jgsource** [reply]
Really much appreciated mate!!!

**Reply by aaylnx** [reply]
Thanks very much for the checkboxes! Much appreciated!

Question. Is there anyway to use these checkboxes on the mobile app? Where there's now ctrl/cmd buttons?

**Comment by Gentleman03** [comment]
@mgmeyers First of all i would like to sa that this plugin is simply amazing and thank you so much for creating it, well done! 

What i miss the most is:

## Breaks with enter
Possibility to make a break with enter. Right now i have to use <br> to make a break and it's such a pain. Maybe shift + enter for break would be ok? Because without break i can't even make lists (ordered and unordered).

## Working TOC for lists
Default table of content in obsidian shows list names correctly but doesn't take you to the list if you click it. It would be great if clicking list name in TOC would bring this list to the center of the screen. It would make navigation through lists waaaay faster it's second most important thing for me after breaks.

## Accepting changes to cards
Click anywhere to enter changes. If you're editing card the only way to accept changes is to click enter. It would be cool if clicking anywhere on board would accept them automatically (exit edition mode).

## Doubl click to edit card
It speaks for itself - double click card to enter edit mode.

## Custom card colors
Having a possibility to change color of a cards would be awesome but it's not as important as for example possibility to make break with enter which is crucial.

**Reply by edward-peters** [reply]
This is a big one for me. It would make it possible to distinguish between say home and work tasks in one lane

**Reply by JeremyNguyenPhD** [reply]
Having the ability to have colored cards would be so useful for me, just to flag some to stand out.

**Reply by JeremyNguyenPhD** [reply]
PS. For anyone looking for the ability to have colored cards,

this isn't quite it, but it's very helpful for me:

we can embed images into cards, eg:

`![](https://www.rsea.com.au/Images/ProductImages/Medium/60268_1.jpg)`

**Comment by PENorris** [comment]
Would be nice to have a quick way to remove a date entry.  Many of my cards get a date for a short time and then go back into a "hold" or "working" status.  Perhaps an entry right under "Edit Date" for "Remove date" makes the most sense?

**Comment by benytzlich** [comment]
Interoperability with calendar and other TODOlists (https://github.com/tasks/tasks)?

**Comment by hepabolu** [comment]
I would like to use colours to differentiate between the cards. 
E.g. the lists in the Kanban show the traditional status (backlog, in progress, done) lists. 

- cards with a deadline can color (more vibrant) red when the deadline approaches
- cards for home projects get a different color than cards for work projects

Alternative: I can add swimlanes to the kanban board (one for home, one for work)

**Reply by promethean** [reply]
Please swimlanes !   For me this is the most important thing.

**Reply by SuperTempAccount** [reply]
+1 for swimlanes 🏊‍♂️ Essential for managing multiple projects. Thank you for all the work developing this plugin 🧡

**Reply by JeremyNguyenPhD** [reply]
+1 for swimlanes
+1 for colors

**Reply by JeremyNguyenPhD** [reply]
PS. For anyone looking for the ability to have colored cards,

this isn't quite it, but it's very helpful for me:

we can embed images into cards, eg:

`![](https://www.rsea.com.au/Images/ProductImages/Medium/60268_1.jpg)`

**Comment by Snuggle** [comment]
# Table View
For me, having the **table view** that is already in the roadmap would really help me with the `databases` feature from moving away from Notion. I would personally love this feature. If it's to be implemented correctly, having colour-coded tags that can autocomplete is quite important to me.


Existing card in roadmap: https://github.com/mgmeyers/obsidian-kanban/projects/1#card-62537721

![image](https://user-images.githubusercontent.com/26250962/133897489-3f1086aa-c825-4dfc-a042-6ae4fe7c0f09.png)

## How do I use the table view?

For my **job applications**, I use a popular template that looks like this:

![image](https://user-images.githubusercontent.com/26250962/133897583-f19805bf-a7d7-45a5-9c57-7e71e2d600df.png)

I also use tables for **keeping track of day-to-day** things too, such as keeping track of certain stats within games, or lists of movies I've watched, places I've been etc.
![image](https://user-images.githubusercontent.com/26250962/133897647-9c30cc17-daee-4855-b57f-54e62b7b6a07.png)

**Comment by davidolrik** [comment]
I would love to be able to use dataview queries to create my board in a fenced markdown block.

Something like this:

```plain
``kanban
query: from "Projects" and #task
lanes: ["Todo", "Blocked", "In progress", "Done"]
done: ["Done"]
``
```

This block would create a board with four lanes, with all the notes in the `Projects` folder which are tagged with `#task`.

- Notes without status will be placed in the first lane.
- Moving a card/note will update the `status` field in the frontmatter of the note.
- Moving to a lane which is also in "done", will allow the card to be archived.
- All meta-data like status, start date, completion date, due date will be contained in the frontmatter of the individual note.

I have created and issue that describes this in slightly more detail #345

**Reply by dwrinn** [reply]
I totally would love a kanban board that works off of a dataview query.

**Comment by longruhai0901** [comment]
Task have endtime，and in a task may need to track several times，so mixed time。。
a sample:
- start  @{2021-12-15}@@{00:00}  
- 1st @{2021-12-16}@@{00:00}  
- 2nd @{2021-12-17}@@{00:00}  
- end @{2021-12-19}@@{00:00}  
the card just display the newest time。

or like this?
yyyymmdd hh:mm1 - yyyymmdd hh:mm2 

Thankyou, i love this plugin more than WeKan, because it can link my all notes and useless labels very cool。

**Comment by wg1k** [comment]
I have some lists I don't use daily, e.g., the Backlog, which I use once per iteration, the Blocked for tasks facing some stopper, and Review for jobs that need approval before completion. 
It would be much appreciated to have the option to toggle the lists (folding or hiding) while not in use to have more screen space for the other lists.

**Comment by simbrett** [comment]
First of all, this is a great plugin - thank you.

I would love to tag the card with the current date whenever it's edited (or moved to a new lane).
That way I can create a dataview query in my daily notes indicating which items I've worked on by day / week etc

**Comment by GGGomer** [comment]
Hi @mgmeyers,

As others have said, great plugin. I started using it recently and it's amazing how polished and good it feels. 

With the new adition of colorful tags I was wondering if it was possible to have the option to give the background and foreground of the entire card the color of the primary tag.

e.g.:

#tag1 has a green background and a light green foreground.
#tag2 has an orange background and a light orange foreground

A card with #tag1 #tag2 would have a green background, and a light green foreground, while the #tag2 tag would still be orange to distinguish the aditional tag. #tag1 would still show, but in the same colors as the card.

**Comment by ingvald** [comment]
new to obsidian, at least in the beginning i need "single source of truth" for my tasks, so i'd like the board to be able to show tasks noted elsewhere in obsidian somehow, like from daily notes, meeting notes, etc. ref https://github.com/mgmeyers/obsidian-kanban/issues/467

**Comment by kevintraywick** [comment]
Love this tool! 
Would like to just see the relative time and date for deadlines, not the @deadlilne{xxxx} stuff, just to keep the board clean. 
Also would be nice to stack lists. Is that vertical swimlanes? Dunno.
Would love to be able to color the background of a card differnt colors. And an option to color all cards in a list the same color.
OK, sorry if these are working now, I'm too busy using your plug-in to watch YT videos :)
Kevin

**Comment by cpyder** [comment]
Thank you so much for this amazing plugin. May I suggest a measure to reduce friction.

Situation:-
I am editing a note and I want to put it into a Kanban board. Currently, I need to open the Kanban board and manually insert it into a list. That is a point of friction.

Proposal:-
What if I could type in a hashtag (eg #ToDo or #Started etc).
The plugin picks up Note Title or Block Reference and places it in the list which is tagged the same (#ToDo or #Started)?

**Comment by sharky98** [comment]
I just started using Kanban today to track some personal projects (and Obsidian few months ago). One friction point I see is the lack of interoperability between Kanban, Tasks and Dataview. While Dataview is more of jack-of-all-trades regarding data queries, I feel that both Kanban and Tasks share a lot of the metadata principles, but not the naming and formatting. It seems both Tasks and Dataview made progress in understanding each other notation (Tasks plugin being [open](https://publish.obsidian.md/tasks/Reference/Task+Formats/Dataview+Format) about this support, while Dataview is [shy about spelling it out loud](https://blacksmithgu.github.io/obsidian-dataview/annotation/metadata-tasks/)), however Kanban notation is left alone.

I see there is already issues regarding this (mainly #548 but #591 and #659 are somewhat related). However, I was not able to find them in the [roadmap](https://github.com/mgmeyers/obsidian-kanban/projects/1), which prompted me to comment here (instead of opening a new issue 😅!).

**Comment by intervolver** [comment]
I think it would be nice if a card could inherit the background color of its tag, and the tag's foreground for the card's border.

**Reply by Rhydderch** [reply]
Yes, this would be awesome!

**Comment by mytmou** [comment]
I wish you could grab and move lists below each other rather than just horizontally. This would be ideal if you have a larger number of lists with a smaller number of cards. 
For example: Let's say you have 10 projects and you want a list of people associated with each project. If it's only 5-6 people per project, there is a lot of free vertical space to utilize. Unless you're using split screen - which is a somewhat complicated/messy way to approach it -  you can, now, only have around 5 lists on your screen at a time before you need to scroll to the side.  If you could use the space below, you could have at least 10, possibly more depending on how long your lists are.

Really enjoying figuring out your program so far though - lots of nifty little things you already thought of!

**Reply by bttger** [reply]
I think tagging/color or swimlanes would be better suited for that purpose.

**Comment by Hi-ImKyle** [comment]
I think my biggest gripe and the main reason why I don't use this plugin often is editing cards is done inline and not with a pop out editor like every kanban board. I know I can make it so the card links back to a note, but I don't like that as it adds a unnecessary transition.

Maybe it's not doable due to limitations in Obsidian, but I'd like to see an actual card editor at some point in the future if it is possible? This would make me drop Trello in a heartbeat..

**Reply by SuperTempAccount** [reply]
Funny, because I'm the complete opposite. 
Card editors feel like they break flow. 
I'd rather be able to edit without opening another window.

**Reply by Hi-ImKyle** [reply]
Always going to be conflicting opinions with how things are done. I get the point of it breaking flow just to edit a card, but I'm not asking to replace what the plugin offers with something completely different. I just would've liked it as an option. Either inline or pop out editor.
